Prox/Prox/Utilities/Future.swift

```diff
@@ -0,0 +1,18 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Deferred
+
+extension FutureProtocol {
+    /// Returns a Future resolved with the result of this Future, or nil
+    /// if this Future isn't resolved by the deadline.
+    func timeout(deadline: DispatchTime) -> Future<Value?> {
```

Comment:
Hm, I haven't run into any problems, though I haven't run this build much. Do you remember what kinds of problems you were seeing? I'd expect maybe compiler errors at worst, but this compiles fine.